### PR TITLE
Results first draft

### PR DESCRIPTION
**Pull request by edudev**
@omarelb @VaniOFX @AmanDaVinci my first draft of the results.
Basically, I just reported the results that I had yesterday.
I shall update the images with the ones Ivan provided.

@AmanDaVinci how's your text about the |Q| timeline going along?

### CODE REVIEW

**Comment by AmanDaVinci** [comment]
I have added the Q Value plot for the mountain car environment. Please add the others too by running:
`python plotting_utils --environment="MountainCar-v0" --q_values`

File: blog_post.md [hunk 1]
```diff
@@ -174,11 +174,70 @@ We use the following hyperparameter settings in all our experiments:
...
+
+After identifying the best hyper-parameters for each of our environments, we
+looked at their divergence in four setups: with all tricks disabled, with
+memory sampling, having a target network, and by having both tricks enabled.
+Our results for MountainCar and Acrobot are shown below
+
+![MountainCar divergence](./img/violinplot_q_divergence_MountainCar-v0_0.99.png)
+![Acrobot divergence](./img/violinplot_q_divergence_Acrobot-v0_0.99.png)
+
+On the y-axis we see a the $$\max |Q|$$ on a logarithmic scale.
```
Comment (edudev) [inline blog_post.md]:
```suggestion
On the y-axis we see the $$\max |Q|$$ on a logarithmic scale.
```

File: blog_post.md [hunk 2]
```diff
@@ -174,11 +174,70 @@ We use the following hyperparameter settings in all our experiments:
...
+// TOOD: maybe think of a theoretical explanation of this?
+
+Lastly, we look at our results on the CartPole environment, for which we
+explicitly disabled the error clipping.
+Mnih et. al. argued that this improved the stability of the neural network,
+however based on our findings, we can say that it also significantly reduced
+the network's learning capacity (and the reward obtained).
+
+![CartPole divergence](./img/violinplot_q_divergence_CartPole-v0_0.99.jpeg)
+// TODO: add reward image of CartPole.
```
Comment (edudev) [inline blog_post.md]:
```suggestion
<> TODO: add reward image of CartPole.
```

File: blog_post.md [hunk 3]
```diff
@@ -174,11 +174,70 @@ We use the following hyperparameter settings in all our experiments:
...
+
+We see that error clipping is instrumental to making the tricks work.
+Without error clipping, the tricks actually hasten the divergence, rather than
+slow it down.
```
Comment (edudev) [inline blog_post.md]:
```suggestion
slow it down in our experiments.
```

File: blog_post.md [hunk 4]
```diff
@@ -174,11 +174,70 @@ We use the following hyperparameter settings in all our experiments:
...
+separate target network help prevent divergence.
+We came to the conclusion that both tricks are in fact necessary and that
+without them the network is likely to spiral out of control within a
+(relatively) short amount of episodes.
+The target network trick proved to be more effective in slowing divergence, but
+it caused less optimal parameters to be learned.
+Combining the two tricks gave us the best results.
+
+We also looked at the effects of clipping the error term during training, and
+understood that netither of tricks is useful without the clipping.
```
Comment (edudev) [inline blog_post.md]:
```suggestion
understood that netither of tricks is useful without the clipping in the simple environments we tested.
```

**Pull request merged**